homeassistant/components/cisco_webex_teams/sensor.py

```diff
@@ -0,0 +1,89 @@
+"""Platform for sensor integration."""
+import logging
+import voluptuous as vol
+from webexteamssdk import WebexTeamsAPI
+
+from homeassistant.const import CONF_TOKEN, CONF_EMAIL, CONF_NAME
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.binary_sensor import (
+    DEVICE_CLASS_PRESENCE,
+    PLATFORM_SCHEMA,
+    Entity
+)
+
+DEFAULT_NAME = "Webex Presence"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
```

Comment:
We no longer accept additions or changes to the yaml config of integrations. See: https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md

This integration needs to be refactored to use a config flow and config entries.

https://developers.home-assistant.io/docs/config_entries_config_flow_handler

Reply:
Thanks Martin I’ll take a look. 👍

Reply:
I recommend using our scaffold script for new integrations.

https://developers.home-assistant.io/docs/creating_component_index

Reply:
Martin do you know if it is possible for multiple integrations in the same folder to share some config such an an auth token? Eg notify and sensor. I don’t see anything in the docs about that. 

Also, is there a process to “upgrade”  the yaml based config to config flow taking the existing values along for the ride?

Reply:
An integration is made up of the config, auth, all platforms (notify and sensor etc) in the same directory.

Here we're creating a webex integration that can have one or multiple platforms, notify, sensor etc.

The auth token will probably be stored in the config entry data. It should be shared among all platforms for the same config entry.

Most entity integrations that support platforms, eg sensor and switch, support forwarding of a config entry. This is how we pass the config entry from the integration `async_setup_entry` to the different platforms `async_setup_entry`. The notify platform doesn't support forwarding though, and we have to fall back to the legacy `async_load_platform` discovery helper. We can store the config in hass.data or pass it via discovery_info to the notify platform.